examples/asrc_demo/src/i2s_audio.c

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2022 XMOS LIMITED. This Software is subject to the terms of the
+// XMOS Public License: Version 1
+
+#define DEBUG_UNIT I2S_AUDIO
+#define DEBUG_PRINT_ENABLE_I2S_AUDIO 1
+
+/* STD headers */
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "rtos_printf.h"
+#include <xcore/hwtimer.h>
+
+/* FreeRTOS headers */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "timers.h"
+#include "queue.h"
+#include "stream_buffer.h"
+
+/* App headers */
+#include "app_conf.h"
+#include "platform/driver_instances.h"
+#include "src.h"
+#include "asrc_utils.h"
+#include "i2s_audio.h"
+#include "rate_server.h"
+#include "tusb_config.h"
+
+
+static void recv_frame_from_i2s(int32_t *i2s_rx_data, size_t frame_count)
+{
+    size_t rx_count =
+    rtos_i2s_rx(i2s_ctx,
+                (int32_t*) i2s_rx_data,
+                frame_count,
+                portMAX_DELAY);
+
+    xassert(rx_count == frame_count);
+
+}
+
+static void i2s_audio_recv_task(void *args)
+{
+    (void)args;
+
+    // Create the 2nd channel ASRC task
+    asrc_process_frame_ctx_t asrc_ctx;
+
+    // 1 ASRC instance per channel, so 2 for 2 channels. Each ASRC instance processes one channel
+    asrc_state_t     asrc_state[NUM_I2S_CHANS][ASRC_CHANNELS_PER_INSTANCE]; //ASRC state machine state
```

Comment:
Yes, I agree, this is not clear. It'll only work with ASRC_CHANNELS_PER_INSTANCE = 1. I'll add a check for now. Since I eventually want to run with ASRC_CHANNELS_PER_INSTANCE = 2, I'll clean it up at that point depending on whether running with ASRC_CHANNELS_PER_INSTANCE is feasible or not.